specs/DevSpecs/telemetry.md

```diff
@@ -0,0 +1,187 @@
+
+### 7.2 JSON Samples and Expected Output 
+ 
+1) 
+``` 
+{ 
+  "type": "AdaptiveCard", 
+  "body": [ 
+    { 
+      "type": "Input.Rating", 
+      "id": "userRating" 
+      } 
+    } 
+  ], 
+  "$schema": "http://adaptivecards.io/schemas/adaptive-card.json", 
+  "version": "1.0" 
+} 
+ 
+``` 
+Expected Values: 
+ 
+| Property                     | Expected Value | 
+|------------------------------|----------------| 
+| CorrelationID                | any            | 
+| AppID                        | any            | 
+| DeviceID                     | any            | 
+| OperationResult              | 0 - Success    | 
+| NumberOfFeedbackControls     | 1              | 
+| NumberOfElements             | 1              | 
+| ElementNames                 | [Input.Rating] | 
+| TotalDefaultRatingScalesUsed | 1              | 
+| TotalDefaultStarImagesUsed   | 1              | 
+| isDeveloper                  | any            | 
+ 
+Note: This information will not be recorded in one `logEvent()` call. Rather, this output reflects the expected values after post-processing of telemetry data.
+ 
+## 8. Estimated Dev Cost 
+With two full-time interns working on this feature, implementation of telemetry in JavaScript should take one month, with estimated delivery being mid-August. Within this time frame, progress in the UWP Adaptive Cards platform may also be a plausible stretch goal. 
+ 
+## 9. JavaScript Architecture Details 
+The main components of telemetry in the JavaScript platform will include the IACLogger wrapper interface as well as the 1DS-specific implementation of the logger interface. Creating a generalized logger interface will allow easy logging to multiple telemetry platforms if desired, as well as prevent reliance of the existing Adaptive Cards code on a specific telemetry platform. The only changes to the existing JavaScript Adaptive Cards code will only include new instantiation and function calls to the generalized IACLogger API.  
+
+### 9.1 ACLogger Telemetry Wrapper 
+The following is the proposed interface for the IACLogger, along with example use cases: 
+
+```
+interface IACLogger { 
+ 
+    logWarn(message: string): void; 
+ 
+    logErr(message: string): void; 
+ 
+    logVerbose(message: string): void; 
+ 
+    logInfo(message: string): void; 
+ 
+    logEvent(event: string, eventSourceName: string, correlationID: string, valueSet?: object): void; 
+ 
+} 
+ 
+enum TelemetryProvider { 
+    OneDataStrategy, 
+    File 
+} 
+ 
+class ACLogger implements IACLogger { 
+    private static instance: IACLogger; 
+    private listProviders: Set<TelemetryProvider>; 
+    private providers: IACLogger[]; 
+ 
+    public static getLogger(): IACLogger { 
+        if (instance) { 
+            return instance; 
+        } else { 
+            // throw error/exception 
+        } 
+    } 
+ 
+    public static setLogger(...providers: TelemetryProvider[]): void { 
```

Comment:
by default we should add console provider i think? Maybe expose a InitializeDefaultProviders() or something